lib/cdo/user_helpers.rb

```diff
@@ -84,6 +84,7 @@ def self.age_from_birthday(birthday)
   # Return the highest age range for the given birthday, e.g.
   # 18+, 13+, 8+ or 4+
   def self.age_range_from_birthday(birthday)
+    return "unknown" unless birthday
     age = age_from_birthday(birthday)
     age_cutoff = AGE_CUTOFFS.find {|cutoff| cutoff <= age}
     age_cutoff ? "#{age_cutoff}+" : nil
```

Comment:
I think it's worth replacing `nil` with `"unknown"` here, so there is no potential bad path remaining (for example if age_from_birthday returned an invalid result). You could even just do that instead of the return statement at the beginning, since it's a simple enough method there isn't a big performance impact.

Reply:
hmmm replacing `nil` with "unknown" here won't work because the error is actually being thrown from `age_from_birthday` which tries to do math with the birthday passed in, which it can't if there isn't a valid value for `birthday`. I think I need the early return to avoid that issue.

Reply:
oh, I see. since the error occurs in `age_from_birthday`, can we handle it there (return nil if the parameter is nil) and then return "unknown" from this method if we don't have a valid `age_cutoff`? In general, I think handling the error as close to where it happens is helpful, so that if the method is used somewhere else, it has the necessary error handling built in.

Reply:
after Erin's changes, it seems like we'd only hit the `nil` condition on this line if the student were less than 4 years old. this might look weird to not have an age, but as long as there's no server error then I think this is ok to ship even though it does leave room for improvement on that unlikely scenario.